pkg/compiler/compilation_document.go

```diff
@@ -21,10 +28,21 @@ type (
 		Configuration    *config.Application
 		Resources        *core.ResourceGraph
 		OutputFiles      []core.File
+		OutputOptions    OutputOptions
+	}
+
+	OutputOptions struct {
+		PostWriteHooks map[string]string `yaml:"post-write-hooks,omitempty"`
 	}
 )
 
+var unquotedCharsRe = regexp.MustCompile(`^[\w.{}:>=<@/-]*$`)
+
 func (doc *CompilationDocument) OutputTo(dest string) error {
+
+	postWriteHooks := newDefaultPostWriteHooksMap()
+	collectionutil.Extend(doc.OutputOptions.PostWriteHooks).Into(postWriteHooks)
```

Comment:
nit (opinion): Even though this is shorter, I don't think it's as clear as written out (and it's only 3 lines and probably the same number of characters overall).